test/ZoraNFTCreatorV1_Fork.t.sol

```diff
@@ -1,70 +1,156 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.10;
 
+import {console2} from "forge-std/console2.sol";
 import {Test} from "forge-std/Test.sol";
 import {IMetadataRenderer} from "../src/interfaces/IMetadataRenderer.sol";
 import "../src/ZoraNFTCreatorV1.sol";
 import "../src/ZoraNFTCreatorProxy.sol";
 import {MockMetadataRenderer} from "./metadata/MockMetadataRenderer.sol";
 import {FactoryUpgradeGate} from "../src/FactoryUpgradeGate.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import {IERC721AUpgradeable} from "erc721a-upgradeable/IERC721AUpgradeable.sol";
 import {ForkHelper} from "./utils/ForkHelper.sol";
-import {DropDeployment , ChainConfig} from "../src/DeploymentConfig.sol";
+import {DropDeployment, ChainConfig} from "../src/DeploymentConfig.sol";
 
-contract ZoraNFTCreatorV1_ForkTests is Test, ForkHelper {
-    address payable public constant DEFAULT_FUNDS_RECIPIENT_ADDRESS =
-        payable(address(0x21303));
+contract ZoraNFTCreatorV1Test is Test, ForkHelper {
+    address public constant DEFAULT_OWNER_ADDRESS = address(0x23499);
+    address payable public constant DEFAULT_FUNDS_RECIPIENT_ADDRESS = payable(address(0x21303));
+    address payable public constant DEFAULT_ZORA_DAO_ADDRESS = payable(address(0x999));
+    address payable public constant mintFeeRecipient = payable(address(0x1234));
+    uint256 public constant mintFee = 0.000777 ether;
+    ERC721Drop public dropImpl;
+    ZoraNFTCreatorV1 public creator;
+    EditionMetadataRenderer public editionMetadataRenderer;
+    DropMetadataRenderer public dropMetadataRenderer;
 
-    function test_fork_CreateEdition() external {
-        string[] memory forkTestChains = getForkTestChains();
+    function makeDefaultSalesConfiguration(uint104 price) internal returns (IERC721Drop.SalesConfiguration memory) {
+        return IERC721Drop.SalesConfiguration({
+                publicSaleStart: 0,
+                publicSaleEnd: type(uint64).max,
+                presaleStart: 0,
+                presaleEnd: 0,
+                publicSalePrice: price,
+                maxSalePurchasePerAddress: 0,
+                presaleMerkleRoot: bytes32(0)
+            });
+    }
 
-        DropDeployment memory deployment;
-        ChainConfig memory chainConfig;
+    function test_create_fork() external {
+        string[] memory forkTestChains = getForkTestChains();
 
         for (uint256 i = 0; i < forkTestChains.length; i++) {
             string memory chainName = forkTestChains[i];
 
-            vm.createSelectFork(vm.rpcUrl(chainName ));
-        
-            // get the deployment for the current chain id.
-            deployment = getDeployment();
-            chainConfig = getChainConfig();
+            vm.createSelectFork(vm.rpcUrl(chainName));
+            creator = ZoraNFTCreatorV1(getDeployment().factory);
+            verifyAddressesFork(chainName);
+            forkEdition();
```

Comment:
it would be better for these each to be separate tests - this way if any of them fails, we get the test reporter reporting the failure of the specific test

Reply:
How can we create separate tests if we don't know the chainNames in advance?

Reply:
oh I see, makes sense.